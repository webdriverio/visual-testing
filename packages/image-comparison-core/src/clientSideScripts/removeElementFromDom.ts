/**
 * Remove a DOM element from the DOM
 */
export default function removeElementFromDom(id: string): void {
    const elem = document.querySelector(`#${id}`)
    if (elem) {
        elem.parentNode!.removeChild(elem)
    }
}
