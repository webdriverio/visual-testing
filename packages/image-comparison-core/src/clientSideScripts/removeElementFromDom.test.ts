// @vitest-environment jsdom

import { describe, it, expect } from 'vitest'
import removeElementFromDom from './removeElementFromDom.js'

describe('removeElementFromDom', () => {
    it('should be able to remove the custom css', () => {
        const id = 'test'
        const cssText = 'body:{width:100%}'
        const head = document.head || document.getElementsByTagName('head')[0]
        const style = document.createElement('style')

        style.id = id
        style.appendChild(document.createTextNode(cssText))
        head.appendChild(style)

        expect(document.head.textContent).toMatchSnapshot()

        removeElementFromDom(id)

        expect(document.head.textContent).toMatchSnapshot()
    })

    it('should do nothing if custom css is not present', () => {
        const id = 'test'

        expect(document.head.textContent).toMatchSnapshot()

        removeElementFromDom(id)

        expect(document.head.textContent).toMatchSnapshot()
    })

    it('should do nothing if document.head is null', () => {
        const id = 'test'
        Object.defineProperty(document, 'head', { value: null })

        removeElementFromDom(id)

        expect(document.head).toBe(null)
    })

    it('should be able to remove an element from the body', () => {
        document.body.innerHTML =
            '<div>' +
            '   <span id="id-1">Hello</span>' +
            '   <span id="id-2">Hello</span>' +
            '   <div>' +
            '     <span id="id-3">Hello</span>' +
            '     <span id="id-4">Hello</span>' +
            '  </div>' +
            '</div>'

        expect(document.body).toMatchSnapshot()

        removeElementFromDom('id-1')

        expect(document.body).toMatchSnapshot()
    })
})
