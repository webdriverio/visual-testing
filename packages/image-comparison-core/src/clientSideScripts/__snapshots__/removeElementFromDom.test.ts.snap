// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`removeElementFromDom > should be able to remove an element from the body 1`] = `
<body>
  <div>
       
    <span
      id="id-1"
    >
      Hello
    </span>
       
    <span
      id="id-2"
    >
      Hello
    </span>
       
    <div>
           
      <span
        id="id-3"
      >
        Hello
      </span>
           
      <span
        id="id-4"
      >
        Hello
      </span>
        
    </div>
  </div>
</body>
`;

exports[`removeElementFromDom > should be able to remove an element from the body 2`] = `
<body>
  <div>
       
       
    <span
      id="id-2"
    >
      Hello
    </span>
       
    <div>
           
      <span
        id="id-3"
      >
        Hello
      </span>
           
      <span
        id="id-4"
      >
        Hello
      </span>
        
    </div>
  </div>
</body>
`;

exports[`removeElementFromDom > should be able to remove the custom css 1`] = `"body:{width:100%}"`;

exports[`removeElementFromDom > should be able to remove the custom css 2`] = `""`;

exports[`removeElementFromDom > should do nothing if custom css is not present 1`] = `""`;

exports[`removeElementFromDom > should do nothing if custom css is not present 2`] = `""`;
